Sources/Fullscreen/FrontpageSearch/Cells/FrontpageSearchImageResultCollectionViewCell.swift

```diff
@@ -0,0 +1,251 @@
+        trailingIconImageView.isHidden = true
+
+        NSLayoutConstraint.activate([
+            layoutGuide.topAnchor.constraint(equalTo: topAnchor),
+            layoutGuide.leadingAnchor.constraint(equalTo: leadingAnchor, constant: .spacingM),
+            layoutGuide.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -.spacingM),
+            layoutGuide.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -.spacingM),
+
+            remoteImageView.heightAnchor.constraint(equalToConstant: imageAndButtonWidth),
+            remoteImageView.widthAnchor.constraint(equalToConstant: imageAndButtonWidth),
+            remoteImageView.leadingAnchor.constraint(equalTo: layoutGuide.leadingAnchor),
+            remoteImageView.bottomAnchor.constraint(equalTo: layoutGuide.bottomAnchor),
+            remoteImageView.topAnchor.constraint(equalTo: layoutGuide.topAnchor),
+
+            iconImageView.topAnchor.constraint(equalTo: layoutGuide.topAnchor),
+            iconImageView.leadingAnchor.constraint(equalTo: layoutGuide.leadingAnchor),
+            iconImageView.bottomAnchor.constraint(equalTo: layoutGuide.bottomAnchor),
+
+            removeButton.heightAnchor.constraint(equalToConstant: imageAndButtonWidth),
+            removeButton.widthAnchor.constraint(equalToConstant: imageAndButtonWidth),
+            removeButton.trailingAnchor.constraint(equalTo: layoutGuide.trailingAnchor),
+
+            favoriteButton.heightAnchor.constraint(equalToConstant: imageAndButtonWidth),
+            favoriteButton.widthAnchor.constraint(equalToConstant: imageAndButtonWidth),
+            favoriteButton.trailingAnchor.constraint(equalTo: layoutGuide.trailingAnchor),
+
+            trailingIconImageView.heightAnchor.constraint(equalToConstant: 14),
+            trailingIconImageView.widthAnchor.constraint(equalToConstant: 14),
+            trailingIconImageView.trailingAnchor.constraint(equalTo: layoutGuide.trailingAnchor)
+        ])
+
+        layer.insertSublayer(highlightLayer, at: 0)
+    }
+
+    public func configure(with item: FrontpageSearchGroupItem, remoteImageViewDataSource: RemoteImageViewDataSource) {
+        titleLabel.attributedText = item.title
+        titleLabel.textColor = item.titleColor
+        subtitleLabel.text = item.subtitle
+
+        configureLeadingImage(with: item, remoteImageViewDataSource)
+        configureTrailingIcon(for: item.displayType)
+        updateFavoriteButton(isFavorite: item.isFavorite)
+        currentItem = item
+    }
+
+    private func configureLeadingImage(with item: FrontpageSearchGroupItem, _ remoteImageViewDataSource: RemoteImageViewDataSource) {
+        guard let imageUrl = item.imageUrl, !imageUrl.isEmpty else {
+            contentStackView.insertArrangedSubview(iconImageView, at: 0)
+            iconImageView.isHidden = false
+            return
+        }
+        contentStackView.insertArrangedSubview(remoteImageView, at: 0)
+        if item.imageContentModeFill == false {
```

Comment:
this can be simplified to `!item.imageContentModeFill`

Reply:
Made imageContentModeFill required and then I could change it. Much prettier I agree.